### PR TITLE
fix #14548 by providing comprehensive message

### LINKED ISSUE #14548: Pkg.update fails with unhelpful message
I just downloaded the latest master and then trying to do a `Pkg.update()` fails with a very unhelpful message.
```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0-dev+1968 (2016-01-02 23:28 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit e55c22a (0 days old master)
|__/                   |  x86_64-w64-mingw32

julia> Pkg.update()
INFO: Updating METADATA...
ERROR: GitError(Code:EMERGECONFLICT, Class:Checkout, 546 conflicts prevent checkout)
 [inlined code] from libgit2/error.jl:98
 in __checkout_tree#337__ at libgit2/repository.jl:148
```
removing the v0.5 package dir and started on a fresh removed the error. I have no clue what caused it.

### PR DESCRIPTION
**Pull request by wildart**

### CODE REVIEW

File: base/pkg/entry.jl [hunk 1]
```diff
@@ -342,10 +342,18 @@ function update(branch::AbstractString)
...
-            LibGit2.rebase!(repo, "origin/$branch")
+        try
+            LibGit2.fetch(repo)
+            ff_succeeded = LibGit2.merge!(repo, fastforward=true)
+            if !ff_succeeded
+                LibGit2.rebase!(repo, "origin/$branch")
+            end
+        catch err
+            if isa(err, LibGit2.Error.GitError)
+                error("METADATA cannot be updated. Error: $(err.msg) [$(err.class)].\nResolve problems manually in $(Pkg.dir("METADATA")).")
```
Comment (tkelman) [inline base/pkg/entry.jl]: are "class" and "code" one-to-one? EMERGECONFLICT is potentially useful to see too

Comment (wildart) [inline base/pkg/entry.jl]: Non of it really useful, because `rebase` and `merge` could produce the same error code and class.

Comment (tkelman) [inline base/pkg/entry.jl]: If it's not one-to-one, then it's at least additional potentially useful information in debugging exactly what conditions caused the error. Should also probably be throwing `PkgError` here.

Comment (wildart) [inline base/pkg/entry.jl]: This info will be available in the rethrown exception.

Comment (tkelman) [inline base/pkg/entry.jl]: this errors before the rethrow. should be PkgError and contain the code somewhere. maybe showerror on the entire GitError in this expanded message?

Comment (wildart) [inline base/pkg/entry.jl]: This rethrow will terminate `update` and show error to user.

Comment (tkelman) [inline base/pkg/entry.jl]: and this error contains slightly less information than the existing one. namely, it's not showing the error code

Comment (wildart) [inline base/pkg/entry.jl]: Sorry, I didn't get it first time.

Comment (tkelman) [inline base/pkg/entry.jl]: https://github.com/JuliaLang/julia/pull/15952/commits/b1cd696788de6d7b5a868d7680c9ef8318df7cf5 looks good

**Pull request merged**